contract/vm.go

```diff
@@ -194,7 +194,7 @@ func NewVmContext(execCtx context.Context, blockState *state.BlockState, cdb Cha
 		confirmed:       confirmed,
 		isQuery:         query,
 		blockInfo:       bi,
-		service:         C.int(service),
+		service:         C.int(executionMode),
```

Comment:
maybe later the `service` from `vmContext` can also be renamed `executionMode`

Reply:
The field `service` seems to have multiple meaings as commented in contract.go file above. IMO, it is better to refactor the overall logic later time.